tests/cases/fourslash/findReferencesBindingPatternInJsdocNoCrash2.ts

```diff
@@ -0,0 +1,33 @@
+/// <reference path='fourslash.ts' />
+
+// @moduleResolution: node
+
+// @Filename: node_modules/use-query/package.json
+////{
+////  "name": "use-query",
+////  "types": "index.d.ts"
+////}
+// @Filename: node_modules/use-query/index.d.ts
+////declare function useQuery(): {
+////  data: string[];
+////};
+
+// @Filename: node_modules/use-query/package.json
+////{
+////  "name": "other",
+////  "types": "index.d.ts"
+////}
+// @Filename: node_modules/other/index.d.ts
+////interface BottomSheetModalProps {
+////  /**
+////   * A scrollable node or normal view.
+////   * @type null | (({ data: any }?) => any)
```

Comment:
is `null |` the only difference between NoCrash2 and 1? If so, I don't see what value it adds above NoCrash1, especially since the new code doesn't reference Typescript types really.

Reply:
This test case exercises a braceless `@type` which was somewhat important when I started to reduce this and for some moments I couldn't figure out why the crash went away on certain reduction attempts. So I included that here to be extra sure that it's gonna be fixed in such a scenario too